frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter, ModalComponentProps } from '../factory/modal';
+import { Dropdown, PromiseComponent } from '../utils';
+import { k8sPatch, K8sResourceKind } from '../../module/k8s';
+
+class ClusterUpdateModal extends PromiseComponent {
+  public state: ClusterUpdateModalState;
+
+  constructor(public props: ClusterUpdateModalProps) {
+    super(props);
+    this._submit = this._submit.bind(this);
+    this._cancel = props.cancel.bind(this);
+    this._change = this._change.bind(this);
+    this.state.selectedVersion = 0;
+  }
+
+  _submit(e) {
```

Comment:
style nit: We've started to use this syntax, then you can remove `bind(this)` from the constructor. With the suggested state change above and these changes to the methods, you can remove the constructor altogether.

```suggestion
  _submit = (e) => {
```